staging/src/k8s.io/apiserver/pkg/authentication/request/x509/x509.go

```diff
@@ -21,16 +21,41 @@ import (
 	"crypto/x509/pkix"
 	"encoding/asn1"
 	"fmt"
+	"math"
 	"net/http"
+	"time"
 
 	"github.com/golang/glog"
+	"github.com/prometheus/client_golang/prometheus"
 
 	utilerrors "k8s.io/apimachinery/pkg/util/errors"
 	"k8s.io/apimachinery/pkg/util/sets"
 	"k8s.io/apiserver/pkg/authentication/authenticator"
 	"k8s.io/apiserver/pkg/authentication/user"
 )
 
+var clientCertificateExpirationHistogram = prometheus.NewHistogram(
+	prometheus.HistogramOpts{
+		Namespace: "apiserver",
+		Subsystem: "client",
+		Name:      "certificate_expiration_seconds",
+		Help:      "Gauge of the remaining lifetime on the certificate used to authenticate a request.",
+		Buckets: []float64{
+			math.Inf(-1),
```

Comment:
I thought the buckets held the max, so a bucket for negative numbers would be set to 0?